src/repositoryModals.jsx

```diff
@@ -0,0 +1,451 @@
+
+import { Alert } from "@patternfly/react-core/dist/esm/components/Alert";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Checkbox } from '@patternfly/react-core/dist/esm/components/Checkbox';
+import { Flex, FlexItem } from "@patternfly/react-core/dist/esm/layouts/Flex/index.js";
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { Modal } from "@patternfly/react-core/dist/esm/components/Modal";
+import { Text } from "@patternfly/react-core/dist/esm/components/Text";
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { TextArea } from "@patternfly/react-core/dist/esm/components/TextArea";
+import { Select, SelectOption } from "@patternfly/react-core/dist/esm/deprecated/components/Select";
+import { Spinner } from "@patternfly/react-core/dist/esm/components/Spinner/index.js";
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+
+import { FormHelper } from 'cockpit-components-form-helper.jsx';
+import { useDialogs } from "dialogs.jsx";
+
+import * as remotes from './remotes';
+
+const _ = cockpit.gettext;
+
+export const RemoveRepositoryModal = ({ origin, availableRemotes, refreshRemotes }) => {
+    const Dialogs = useDialogs();
+    const [error, setError] = useState('');
+    const [selectedRemotes, setSelectedRemotes] = useState([]);
+
+    const onDelete = () => {
+        Promise.all(selectedRemotes.map(remote => remotes.deleteRemote(remote)))
+                .then(() => refreshRemotes())
+                .then(Dialogs.close, ex => setError(ex.message));
+    };
+
+    const handleChange = (event, isChecked) => {
+        if (isChecked) {
+            setSelectedRemotes([...selectedRemotes, event.target.id]);
+        } else {
+            setSelectedRemotes(selectedRemotes.filter(remote => remote !== event.target.id));
+        }
+    };
+
+    const actions = [
+        <Button key="remove-repo"
+            variant="danger"
+            isDisabled={selectedRemotes.length === 0}
+            onClick={() => onDelete()}>
+            {_("Remove")}
+        </Button>,
+        <Button key="cancel" variant="link" onClick={Dialogs.close}>
+            {_("Cancel")}
+        </Button>
+    ];
+
+    const repositories = availableRemotes.map(remote => {
+        return (
+            <Checkbox label={remote}
+                key={remote}
+                id={remote}
+                onChange={handleChange}
+                isChecked={selectedRemotes.includes(remote)}
+                isDisabled={origin.remote === remote} />
+        );
+    });
+
+    return (
+        <Modal isOpen
+            id="remove-repository-modal"
+            title={_("Remove repository")}
+            position="top"
+            variant="small"
+            onClose={Dialogs.close}
+            actions={actions}
+        >
+            <Form isHorizontal>
+                {error && <Alert variant="danger" isInline title={error} />}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231124-162305-47d51597-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)